dev-tools/devmachine/devmachine.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package devmachine
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	compute "google.golang.org/api/compute/v1"
+)
+
+func Run(instanceName string, imageName string, zone string) error {
+	ctx := context.Background()
+
+	projectID := "elastic-platform-ingest"
+	log.Println("Authenticating with GCP...")
+	computeService, err := compute.NewService(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create compute service: %v. \nRun `gcloud auth login` to make it work", err)
+	}
+
+	imageProject := "elastic-images-prod"
+	machineType := fmt.Sprintf("zones/%s/machineTypes/n1-standard-4", zone)
+	sourceImage := fmt.Sprintf("projects/%s/global/images/%s", imageProject, imageName)
+
+	instance := &compute.Instance{
+		Name:        instanceName,
+		MachineType: machineType,
+		Disks: []*compute.AttachedDisk{
+			{
+				AutoDelete: true,
+				Boot:       true,
+				InitializeParams: &compute.AttachedDiskInitializeParams{
+					SourceImage: sourceImage,
+				},
+			},
+		},
+		NetworkInterfaces: []*compute.NetworkInterface{
+			{
+				AccessConfigs: []*compute.AccessConfig{
+					{
+						Name: "External NAT",
+						Type: "ONE_TO_ONE_NAT",
+					},
+				},
+				Network: "global/networks/default", // Use default network
+			},
+		},
+		Tags: &compute.Tags{
+			Items: []string{"http-server", "https-server"},
```

Comment:
Why this tag specifically? Shouldn't we have some tags here identifying this instance as a dev machine as well?

Reply:
Good catch! Replaced tags with required labels